tsdb/agent/storage_test.go

```diff
@@ -0,0 +1,325 @@
+// Copyright 2017 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package agent
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/promlog"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/storage/remote"
+	"github.com/prometheus/prometheus/tsdb/tsdbutil"
+	"github.com/stretchr/testify/require"
+)
+
+var promAgentData = filepath.Join(os.TempDir(), "data-agent")
+
+func TestQuerier(t *testing.T) {
+	opts := DefaultOptions()
+	cfg := promlog.Config{}
+	logger := promlog.New(&cfg)
+
+	s, err := NewStorage(logger, prometheus.DefaultRegisterer, nil, promAgentData, opts)
+	if err != nil {
+		t.Fatalf("unable to create storage for the agent: %v", err)
+	}
+
+	_, err = s.Querier(context.TODO(), 0, 0)
+	s.Close()
+	require.Equal(t, err, ErrUnsupported)
+}
```

Comment:
You could probably reduce the repetition here with some subtests:

```go
func TestUnsupported(t *testing.T) {
	opts := DefaultOptions()
	cfg := promlog.Config{}
	logger := promlog.New(&cfg)

	s, err := NewStorage(logger, prometheus.DefaultRegisterer, nil, promAgentData, opts)
	if err != nil {
		t.Fatalf("unable to create storage for the agent: %v", err)
	}
	defer s.Close()
	
	t.Run("Querier", func(t *testing.T) {
	        _, err := s.Querier(context.TODO(), 0, 0)
	        require.Equal(t, err, ErrUnsupported)
	})

       // t.Run("ChunkQuerier", func(t *testing.T) {
       // t.Run("ExemplarQuerier", func(t *testing.T) {
}
```

Reply:
👍